packages/key-management/src/util/key.ts

```diff
@@ -26,15 +34,17 @@ export interface DeriveAccountPrivateKeyProps {
   rootPrivateKey: Crypto.Bip32PrivateKeyHex;
   accountIndex: number;
   bip32Ed25519: Crypto.Bip32Ed25519;
+  purpose?: KeyPurpose;
 }
 
 export const deriveAccountPrivateKey = async ({
   rootPrivateKey,
   accountIndex,
-  bip32Ed25519
+  bip32Ed25519,
+  purpose = KeyPurpose.STANDARD
```

Comment:
We may need to also update hardware wallet key agents to take variable purpose:

https://github.com/input-output-hk/cardano-js-sdk/blob/fac71bd000a4fc4234373674e127960f3f8b500b/packages/key-management/src/util/key.ts#L58-L90

and here:

```
  static async getXpub({
    deviceConnection,
    communicationType,
    accountIndex
  }: GetLedgerXpubProps): Promise<Crypto.Bip32PublicKeyHex> {
    try {
      const recoveredDeviceConnection = await LedgerKeyAgent.checkDeviceConnection(communicationType, deviceConnection);
      const derivationPath = `${CardanoKeyConst.PURPOSE}'/${CardanoKeyConst.COIN_TYPE}'/${accountIndex}'`;
      const extendedPublicKey = await recoveredDeviceConnection.getExtendedPublicKey({
        path: str_to_path(derivationPath) // BIP32Path
      });
      const xPubHex = `${extendedPublicKey.publicKeyHex}${extendedPublicKey.chainCodeHex}`;
      return Crypto.Bip32PublicKeyHex(xPubHex);
    } catch (error: any) {
      if (error.code === 28_169) {
        throw new errors.AuthenticationError('Failed to export extended account public key', error);
      }
      throw transportTypedError(error);
    }
  }
  ```
  
  But I am not sure if hardware wallet would support a different purpose for the key path, we need to test this

Reply:
To generalize this change request: we need to review usages of every `CardanoKeyConst.PURPOSE`

![image](https://github.com/input-output-hk/cardano-js-sdk/assets/2457868/6207b978-b2a5-4ae2-835a-d31e6d6f44e1)

Reply:
Regarding HW wallets with 1854 purpose, maybe @tomislavhoracek can advise

Reply:
I allowed for a variable purpose in derivation paths on 5d8313b

Reply:
1854 keys with HW might require a different [signing mode](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/hardware-ledger/src/LedgerKeyAgent.ts#L502). We need to test it to be sure.

Reply:
Also, you seem to be just omitting `purpose` in many places ([example](https://github.com/input-output-hk/cardano-js-sdk/commit/5d8313b8232afab29e07b40890a86a08688b44e0#diff-39763719dec2cf7895e133f8f4b8f7a2ce0ed4a861a3c57d2efef4d8db919be9R107)), which means it will always use 1852 as it's the default. That's probably not what we want.

Reply:
Yes I think that is a good idea

Reply:
@mkazlauskas @greatertomi Ledger `CardanoKeyConst.PURPOSE = 1854` checked and it works.
Here are results 
![Screenshot 2024-05-29 at 20 29 59](https://github.com/input-output-hk/cardano-js-sdk/assets/18653950/1f107fe3-5652-42d5-9d5e-07e7d8969307)
![IMG_4234](https://github.com/input-output-hk/cardano-js-sdk/assets/18653950/18ad463b-b730-4898-8a26-596454b19089)

p.s. you can run this test to check your implementation with different "purpose"  ( command: `yarn test:hw:ledger`)
https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/wallet/test/hardware/ledger/LedgerKeyAgent.integration.test.ts#L51

Reply:
Have you tested signing with 1854 keys @tomislavhoracek ?

Reply:
@mkazlauskas no, I just quickly checked is a different purpose whitelisted on Ledger by exporting xpub. 
But, I can check it, of course

Reply:
Ok, I tried. I'm able to sign tx with device but mapping fails with msg:
 HwMappingError: Hardware data mapping failure: Ledger computed a different transaction idLedgerKeyAgent.signTransaction`

Reply:
Is that error SDK-related that can be resolved by us?

Reply:
enforced derivation path on 82449e7